libkbfs/block_retrieval_worker.go

```diff
@@ -48,12 +48,12 @@ func newBlockRetrievalWorker(bg blockGetter,
 // blockGetter.getBlock, and responds to the subscribed requestors with the
 // results.
 func (brw *blockRetrievalWorker) HandleRequest() (err error) {
-	brw.queue.Work(brw.workCh)
 	var retrieval *blockRetrieval
 	select {
-	case retrieval = <-brw.workCh:
+	case <-brw.workCh:
+		retrieval = brw.queue.popIfNotEmpty()
```

Comment:
1. 100 on-demand requests come in.
2. 2 prefetch requests come in.
3. The 2 prefetch workers pick up 2 of the on-demand requests.
4. 98 on-demand requests and 2 prefetch requests are picked by on-demand requests.
5. The prefetch workers finish.
6. 2 more on-demand requests come in.
7. 2 prefetch requests come in.
8. The 2 prefetch workers pick up the 2 on-demand requests.
9. 2 of the on-demand workers free up from servicing on-demand requests and pick up the 2 prefetch requests.
10. At this point, we now have 4 simultaneous prefetch requests in flight, using up CPU and network that we'd rather be using for the on-demand requests.  The point of this patch is to limit the amount of CPU and network used by prefetching, right?
11. Repeat this pattern over and over, until all on-demand workers are doing prefetching, while new on-demand requests block.